src/scripting_engine.h

```diff
@@ -0,0 +1,72 @@
+#ifndef _ENGINE_H_
+#define _ENGINE_H_
+
+#include "server.h"
+
+// Forward declaration of the engine structure.
+typedef struct scriptingEngine scriptingEngine;
+
+/* ValkeyModule type aliases for scripting engine structs and types. */
+typedef ValkeyModuleScriptingEngineCtx engineCtx;
+typedef ValkeyModuleScriptingEngineFunctionCtx functionCtx;
+typedef ValkeyModuleScriptingEngineCompiledFunction compiledFunction;
+typedef ValkeyModuleScriptingEngineMemoryInfo engineMemoryInfo;
+typedef ValkeyModuleScriptingEngineMethods engineMethods;
+
+/*
+ * Callback function used to iterate the list of engines registered in the
+ * engine manager.
+ *
+ * - `engine`: the scripting engine in the current iteration.
+ *
+ * - `context`: a generic pointer to a context object.
+ *
+ * If the callback function returns 0, then the iteration is stopped
+ * immediately.
+ */
+typedef int (*engineIterCallback)(scriptingEngine *engine, void *context);
+
+/*
+ * Engine manager API functions.
+ */
+int engineManagerInit(void);
+size_t engineManagerGetCacheMemory(void);
+size_t engineManagerGetNumEngines(void);
+size_t engineManagerGetMemoryUsage(void);
+int engineManagerRegisterEngine(const char *engine_name,
```

Comment:
I suggest prefixing the names with `scriptingEngine`.

Maybe we can avoid some repetition of "engine" and make the "manager" part be implicit, at least in some cases? Isn't it clear enough from the parameters that a function is for one engine or for the entire subsystem (manager)?

Idea:

```suggestion
int scriptingEngineRegister(const char *engine_name,
```